tests/run/i11050.scala

```diff
@@ -0,0 +1,131 @@
+import scala.compiletime.*
+import scala.deriving.*
+
+object OriginalReport:
+  sealed trait TreeValue
+  sealed trait SubLevel extends TreeValue
+  case class Leaf1(value: String) extends TreeValue
+  case class Leaf2(value: Int)    extends SubLevel
+
+// Variants from the initial failure in akka.event.LogEvent
+object FromAkkaCB:
+  sealed trait A
+  sealed trait B extends A
+  sealed trait C extends A
+  case class D() extends B, C
+  case class E() extends C, B
+
+object FromAkkaCB2:
+  sealed trait A
+  sealed trait N extends A
+  case class B() extends A
+  case class C() extends A, N
+
+object FromAkkaCB3:
+  sealed trait A
+  case class B() extends A
+  case class C() extends A
+  class D extends C // ignored pattern: class extending a case class
+
+object FromAkkaCB4:
+  sealed trait LogEvent
+  object LogEvent
+  case class Error() extends LogEvent
+  class Error2() extends Error() with LogEventWithMarker // ignored pattern
+  case class Warning() extends LogEvent
+  sealed trait LogEventWithMarker extends LogEvent // must be defined late
+
+object FromAkkaCB4simpler:
+  sealed trait LogEvent
+  object LogEvent
+  case class Error() extends LogEvent
+  class Error2() extends LogEventWithMarker // not a case class
+  case class Warning() extends LogEvent
+  sealed trait LogEventWithMarker extends LogEventWithMarker
```

Comment:
this is causing an error due to the cyclic inheritance
```suggestion
  sealed trait LogEventWithMarker extends LogEvent
```